app/commands/v2/republish.rb

```diff
@@ -0,0 +1,74 @@
+module Commands
+  module V2
+    class Republish < BaseCommand
+      delegate :content_id, :locale, to: :document
+
+      def call
+        validate
+        republish_edition
+        after_transaction_commit { send_downstream } if downstream
+
+        Success.new(content_id: content_id)
+      end
+
+    private
+
+      def document
```

Comment:
Not related to this PR, but I feel like we could move this into `BaseCommand`.

Reply:
I don't think commands are specific to documents unfortunately and similarly it's a bit of "where did that come from inheritance issue" :-/